File: exercism/java/space-age/src/main/java/SpaceAge.java
class SpaceAge {
    enum SolarSystem {
        MERCURY(0.2408467f),
        VENUS(0.61519726f),
        EARTH(1),
        MARS(1.8808158f),
        JUPITER(11.862615f),
        SATURN(29.447498f),
        URANUS(84.016846f),
        NEPTUNE(164.79132f);

        private static int EARTH_YEAR_IN_SECONDS = 31557600;
        private final double orbitalPeriodInEarthYears;

        SolarSystem(double earthYears) {
            this.orbitalPeriodInEarthYears = earthYears;
        }

        double localAge(long seconds) {
            return (double) seconds /EARTH_YEAR_IN_SECONDS / orbitalPeriodInEarthYears;
        }
    }

    private long secondsOnEarth;

    SpaceAge(long seconds) {
        this.secondsOnEarth = seconds;
    }

    long getSeconds() {
        return secondsOnEarth;
    }

    double onEarth() {
        return SolarSystem.EARTH.localAge(secondsOnEarth);
    }

    double onMercury() {
        return SolarSystem.MERCURY.localAge(secondsOnEarth);
    }

    double onVenus() {
        return SolarSystem.VENUS.localAge(secondsOnEarth);
    }

    double onMars() {
        return SolarSystem.MARS.localAge(secondsOnEarth);
    }

    double onJupiter() { return SolarSystem.JUPITER.localAge(secondsOnEarth); }

    double onSaturn() {
        return SolarSystem.SATURN.localAge(secondsOnEarth);
    }

    double onUranus() {
        return SolarSystem.URANUS.localAge(secondsOnEarth);
    }

    double onNeptune() { return SolarSystem.NEPTUNE.localAge(secondsOnEarth); }
}

File: exercism/python/space-age/space_age.py
class SpaceAge:
    SECONDS_IN_EARTH_YEAR = 31_557_600

    to_earth = {
        "mercury": 0.2408467,
        "venus": 0.61519726,
        "earth": 1,
        "mars": 1.8808158,
        "jupiter": 11.862615,
        "saturn": 29.447498,
        "uranus": 84.016846,
        "neptune": 164.79132
    }

    def __init__(self, seconds):
        self.seconds = seconds

    def on_mercury(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["mercury"], 2)

    def on_venus(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["venus"], 2)

    def on_earth(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["earth"], 2)

    def on_mars(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["mars"], 2)

    def on_jupiter(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["jupiter"], 2)

    def on_saturn(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["saturn"], 2)

    def on_uranus(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["uranus"], 2)

    def on_neptune(self):
        return round(self.__seconds_to_year__() / SpaceAge.to_earth["neptune"], 2)

    def __seconds_to_year__(self):
        return self.seconds / SpaceAge.SECONDS_IN_EARTH_YEAR
